### PR TITLE
Deprecate `Util.copyFile()`

### PR DESCRIPTION
**Pull request by StefanSpieker**
Deprecate `copyFile` which is not needed anymore. I thought about replacing the Ant code with a Java 7 api function, but actually the whole functionality is quite obsolete. I was not able to find a usage with the GitHub search. But with taht I can take a look at the plugin usage output.

### Proposed changelog entries

* Internal: Deprecate `Util.copyFile()`

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by timja — APPROVED** [review]

**Review by basil — APPROVED** [review]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

File: core/src/main/java/hudson/Util.java
Comment (StefanSpieker) [reply core/src/main/java/hudson/Util.java:1079]: The import should be there already

**Review by oleg-nenashev — APPROVED** [review]
We may merge it in 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process

File: core/src/main/java/hudson/Util.java
Comment (jglick) [inline core/src/main/java/hudson/Util.java:1079]: Please do not add `@Restricted` on top of `@Deprecated`. It is superfluous and can cause problems in cross-testing.

Comment (timja) [reply core/src/main/java/hudson/Util.java:1079]: the point of this pattern is to force consumers to update when they move their core version so it doesn't stay around forever. 

what's the problem in testing it causes? you could always run with ignoring restrictions for that

Comment (jglick) [reply core/src/main/java/hudson/Util.java:1079]: > force consumers to update

There are a lot of things that are `@Deprecated` that consumers _should_ update as time permits as part of routine maintenance. If something is “worse than deprecated”—actively harmful and must be deleted soon in order to unblock something else—then seek out and fix callers. In this case there is no particular harm that I can see (unless and until we want to remove/detach Ant) so this annotation is just a nuisance.

The purpose of `@Restricted` is to mark a language element which must be `public` for some technical reason but which was never intended to be referred to from Java code, or only from Java code within the current project, etc.

**Pull request merged**
